packages/experiment-core/src/api/stream-api.ts

```diff
@@ -0,0 +1,194 @@
+import {
+  StreamEventSource,
+  StreamEventSourceClass,
+  StreamErrorEvent,
+  StreamMessageEvent,
+  DefaultStreamErrorEvents,
+} from '../transport/stream';
+
+const DEFAULT_INITIAL_CONN_TIMEOUT = 1000; // Initial connection timeout.
+const MAX_CONN_MS_MIN = 12 * 60 * 1000; // Min of max connection timeout and wants to automatically disconnect and reconnects.
+const MAX_CONN_MS_MAX = 18 * 60 * 1000; // Max of the above timeout.
+const KEEP_ALIVE_INTERVAL = (15 + 2) * 1000; // 15 seconds plus 2 seconds grace period. // 0 or neg value disables keep alive.
+const KEEP_ALIVE_DATA = ' ';
+
+export type StreamOptions = {
+  libraryName: string;
+  libraryVersion: string;
+};
+export type StreamOnUpdateCallback = (data: string) => unknown;
+export type StreamOnErrorCallback = (err: StreamErrorEvent) => unknown;
+
+export interface StreamApi {
```

Comment:
Also, one-off: This seems like a transform interface and implementation to me, not necessarily an API.

I would consider moving this into the `transport/streatm.ts` file.